### PR TITLE
Tetgen fix + tests

### PR DESCRIPTION
**Pull request by rinkk**

### CODE REVIEW

**Comment by rinkk** [comment]
test this please!

File: Tests/FileIO/TestTetGenInterface.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	std::vector<MeshLib::Node> attr_pnts;
+	FileIO::TetGenInterface tgi;
+	bool result (tgi.writeTetGenSmesh(output_name, *mesh, attr_pnts));
+	ASSERT_TRUE(result);
+
+	GeoLib::GEOObjects geo_objects;
+	result = tgi.readTetGenGeometry(output_name, geo_objects);
+	ASSERT_TRUE(result);
+	std::string const ref_name(BaseLib::BuildInfo::data_path + "/FileIO/AmmerSubsurfaceCoarse.smesh");
+	result = tgi.readTetGenGeometry(ref_name, geo_objects);
```
Comment (TomFischer) [inline Tests/FileIO/TestTetGenInterface.cpp]: I wonder why `readTetGenGeometry` is called here a second time. The first call is in line 60. Can you comment this?

File: Tests/FileIO/TestTetGenInterface.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	ASSERT_TRUE(result);
+
+	std::vector<GeoLib::Point*> const& ref_pnts (*geo_objects.getPointVec("AmmerSubsurfaceCoarse"));
+	std::vector<GeoLib::Point*> const& new_pnts (*geo_objects.getPointVec(tg_new_name));
+	ASSERT_EQ(ref_pnts.size(), new_pnts.size());
+
+	std::vector<GeoLib::Surface*> const& ref_sfc (*geo_objects.getSurfaceVec("AmmerSubsurfaceCoarse"));
```
Comment (TomFischer) [inline Tests/FileIO/TestTetGenInterface.cpp]: Why the line break?

File: Tests/FileIO/TestTetGenInterface.cpp (hunk 1 above)
Comment (rinkk) [inline Tests/FileIO/TestTetGenInterface.cpp]: The first call reads written output from the mesh conversion (```output_name```), this one reads the reference files (```ref_name```) with which the output is compared.

Comment (TomFischer) [inline Tests/FileIO/TestTetGenInterface.cpp]: Thank you, I understand.

**Pull request merged**